# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/NativeLayerCA.mm
# Commit: 98460b4d7285
# Full Hash: 98460b4d7285e53cc94858bcef9b44cafc003c3a
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2022-07-22 03:25:11
# Description:
#   Bug 1780586: Make IsCGColorOpaqueBlack handle clear color correctly. r=gfx-reviewers,jrmuizel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152439
# ==============================================================================

diff -r 951002217e9f -r 98460b4d7285 gfx/layers/NativeLayerCA.mm
--- a/gfx/layers/NativeLayerCA.mm	Thu Jul 21 17:55:45 2022 +0000
+++ b/gfx/layers/NativeLayerCA.mm	Thu Jul 21 18:02:34 2022 +0000
@@ -498,6 +498,11 @@
     return true;
   }
   size_t componentCount = CGColorGetNumberOfComponents(aColor);
+  if (componentCount == 0) {
+    // This will happen if aColor is kCGColorClear. It's not opaque black.
+    return false;
+  }
+
   const CGFloat* components = CGColorGetComponents(aColor);
   for (size_t c = 0; c < componentCount - 1; ++c) {
     if (components[c] > 0.0f) {
